recipes/libsodium/all/conanfile.py

```diff
@@ -28,54 +32,100 @@ class LibsodiumConan(ConanFile):
         "PIE": False,
     }
 
-    short_paths = True
-
-    _autotools = None
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
-    @property
-    def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
+
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
```

Comment:
Why these new empty lines? For a class' method, only one empty line is enough. Try following the template.

Reply:
I find it harder to read if there is just one line between methods.  I can remove.

Reply:
please, keep only one line for class methods. https://peps.python.org/pep-0008/#blank-lines